verticapy/ml_pipeline/_metrics.py

```diff
@@ -0,0 +1,1865 @@
+#
+# VerticaPy aims to do all of the above. The idea is simple: instead of moving
+# data around for processing, VerticaPy brings the logic to the data.
+#
+#
+# Modules
+#
+# Standard Python Modules
+import math
+from collections.abc import Iterable
+from typing import Union
+
+# Other Modules
+import numpy as np
+
+# VerticaPy Modules
+from verticapy import *
+from verticapy import vDataFrame
+from verticapy.machine_learning.model_selection import *
+from verticapy._utils._sql._sys import _executeSQL
+from verticapy._utils._sql._collect import save_to_query_profile
+from ._toolbox import *
+
+# from verticapy.toolbox import *
+
+
+#
+# Function used to simplify the code
+#
+# ---#
+def compute_metric_query(
+    metric: str,
+    y_true: str,
+    y_score: str,
+    input_relation: Union[str, vDataFrame],
+    title: str = "",
+    fetchfirstelem: bool = True,
+):
+    """
+    ---------------------------------------------------------------------------
+    A helper function that uses a specified metric to generate and score a query.
+
+    Parameters
+    ----------
+    metric: str
+        The metric to use in the query.
+    y_true: str
+        Response column.
+    y_score: str
+        Prediction.
+    input_relation: str/vDataFrame
+        Relation to use for scoring. This relation can be a view, table, or a
+        customized relation (if an alias is used at the end of the relation).
+        For example: (SELECT ... FROM ...) x
+    title: str, optional
+        Relation to use to do the scoring. The relation can be a view or a table
+        or even a customized relation. For example, you could write:
+        "(SELECT ... FROM ...) x" as long as an alias is given at the end of the
+    Title of the query.
+    fetchfirstelem: bool, optional
+        If set to True, this function returns one element. Otherwise, this
+        function returns a tuple.
+
+    Returns
+    -------
+    float or tuple of floats
+        score(s)
+    """
+    check_types(
+        [
+            ("metric", metric, [str]),
+            ("y_true", y_true, [str]),
+            ("y_score", y_score, [str]),
+            ("input_relation", input_relation, [str, vDataFrame]),
+            ("title", title, [str]),
+            ("fetchfirstelem", fetchfirstelem, [bool]),
+        ]
+    )
+    relation = (
+        input_relation
+        if isinstance(input_relation, str)
+        else input_relation.__genSQL__()
+    )
+    query = "SELECT /*+LABEL('learn.metrics.compute_metric_query')*/ {0} FROM {1} WHERE {2} IS NOT NULL AND {3} IS NOT NULL;".format(
+        metric.format(y_true, y_score), relation, y_true, y_score
+    )
+    return query  # Return the query
+    # return executeSQL(
+    #     query, title=title, method="fetchfirstelem" if fetchfirstelem else "fetchrow"
+    # )
+
+
+# ---#
+def compute_tn_fn_fp_tp(
+    y_true: str,
+    y_score: str,
+    input_relation: Union[str, vDataFrame],
+    pos_label: Union[int, float, str] = 1,
+):
+    """
```

Comment:
this file seems to be an old version of metrics. Please remove it and important directly the right functions.